components/core/src/clp/streaming_archive/ArchiveMetadata.hpp

```diff
@@ -79,15 +96,46 @@ class ArchiveMetadata {
 
     [[nodiscard]] auto get_end_timestamp() const { return m_end_timestamp; }
 
+    [[nodiscard]] auto get_variable_encoding_methods_version() const -> std::string_view const& {
+        return m_variable_encoding_methods_version;
+    }
+
+    [[nodiscard]] auto get_variables_schema_version() const -> std::string_view const& {
+        return m_variables_schema_version;
+    }
+
+    [[nodiscard]] auto get_compression_type() const -> std::string_view const& {
+        return m_compression_type;
+    }
+
     /**
      * Expands the archive's time range based to encompass the given time range
      * @param begin_timestamp
      * @param end_timestamp
      */
     void expand_time_range(epochtime_t begin_timestamp, epochtime_t end_timestamp);
 
+    /**
+     * Packs `ArchiveMetadata` to MessagePack and writes to the open file.
+     *
+     * @param file_writer Writer for archive metadata file.
+     * @throw FileWriter::OperationFailed if failed to write.
```

Comment:
```suggestion
```

Generally we don't document thrown exceptions, except at the boundary of a library, for the following reasons:

* C++ exceptions propagate through functions, so tracing every exception that can be thrown to document it is impractical. The best you could do is catch in every method and then re-throw (this could be the strategy to use at the boundary of a library's interface).
* As mentioned in the other comment, C++ exceptions should be used in exceptional circumstances---this is so that callers don't need to be expected to gracefully handle every exception (instead, they can simply log the error and exit).